File: proyectomovil/lib/models/inventario.dart
// lib/models/inventario.dart
class Inventario {
  final int idInventario;
  final int idDetalleProducto;
  final String nombreProducto;
  final String marca;
  final String modelo;
  final String categoria;
  final double precioUnitario;
  int cantidad; // editable para la venta
  final bool estado;

  Inventario({
    required this.idInventario,
    required this.idDetalleProducto,
    required this.nombreProducto,
    required this.marca,
    required this.modelo,
    required this.categoria,
    required this.precioUnitario,
    required this.cantidad,
    required this.estado,
  });

  factory Inventario.fromJson(Map<String, dynamic> json) {
    return Inventario(
      idInventario: json['idInventario'],
      idDetalleProducto: json['idDetalleProducto'],
      nombreProducto: json['nombreProducto'] ?? '',
      marca: json['marca'] ?? '',
      modelo: json['modelo'] ?? '',
      categoria: json['categoria'] ?? '',
      precioUnitario: (json['precioUnitario'] ?? 0).toDouble(),
      cantidad: json['cantidad'] ?? 0,
      estado: json['estado'] ?? true,
    );
  }
}

File: ProjectIV-main/WebApi.Model/Inventario.cs
namespace WebApi.Model
{

    public class Inventario
    {
        public int IdInventario { get; set; }
        public int IdDetalleProducto { get; set; }
        public string NombreProducto { get; set; }
        public string Marca { get; set; }
        public string Modelo { get; set; }
        public string Categoria { get; set; }
        public decimal PrecioUnitario { get; set; }
        public int Cantidad { get; set; }
        public bool Estado { get; set; }
        public DateTime FechaRegistro { get; set; }
    }

}